service_crategen/src/commands/generate/codegen/rest_json.rs

```diff
@@ -113,11 +113,15 @@ impl GenerateProtocol for RestJsonGenerator {
         Ok(())
     }
 
-    fn generate_prelude(&self, writer: &mut FileWriter, _: &Service) -> IoResult {
-        writeln!(writer,
-                 "use serde_json;
-        use rusoto_core::param::{{Params, ServiceParams}};
-        use rusoto_core::signature::SignedRequest;
+    fn generate_prelude(&self, writer: &mut FileWriter, service: &Service) -> IoResult {
+        writeln!(writer, "use serde_json;")?;
+
+        // avoid unused imports when building services that don't use params
+        if service.service_type_name() != "Batch" {
```

Comment:
As I've been adding a few more of the `rest-json` services, I've found a few more that also need this import removed.

Instead of checking by a hard-coded list of service names, I can just write a function that recurses through the input shapes and determines if the import is actually necessary for any given service.